Tasks/authenticode-sign/entry.ts

```diff
@@ -1,38 +1,47 @@
-import * as tl from "vsts-task-lib/task";
 import * as tr from "vsts-task-lib/ToolRunner";
+import * as tl from "vsts-task-lib/task";
+import * as sf from "./SecureFileHelpers";
 
+let secureFileId: string = null;
+let secureFileHelpers: sf.SecureFileHelpers = null;
 async function run() {
     let signToolLocation: string = getSignToolLocation();
     let retryCount: number = Number(tl.getInput("retryCount", true));
 
     let signtool: tr.ToolRunner = new tr.ToolRunner(signToolLocation);
-    
+
     let signtoolArguments: string[] = ["sign"];
 
     pushTimestampArgs(signtoolArguments);
-    pushCertArgs(signtoolArguments);
-    pushFileArgs(signtoolArguments);    
-
-    signtool.arg(signtoolArguments);
-
-    let i: number = 0;
-    
-    // sometime timestamp servers don't reponse
-    while(true) {
-        try
-        {
-            var exitCode: number = await signtool.exec();
-            tl.debug(`Exit code: ${exitCode}`);
-            break;
-        }
-        catch(e) {
-            console.log(`Error attempting to sign. Attempt number: ${i++}. Exception text: ${e}`);
-            if (i == retryCount) {
-                tl.setResult(tl.TaskResult.Failed, `Unable to sign ${e}`);
-                throw e;
+    try {
+        pushCertArgs(signtoolArguments);

```

Comment:
you should await pushCertArgs, otherwise the certifate gets downloaded too late, and signing fails

Reply:
Thanks! Can't seem to make tslint pick that up unfortunately. Seems like something that should be a basic warning.